### PR TITLE
Gen masks

### LINKED ISSUE #1: Read annotation format into Shapely objects
**Is your feature request related to a problem? Please describe.**
The annotation output of SlideScore has to be parsed into proper objects.

**Describe the solution you'd like**
Export objects into Shapely objects (or similar).

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by AjeyPaiK**
Fixes #1, #3, #4

Try 

`slidescore -s 638 write-shapely ./TISSUE_COMPARTMENTS_21_10_19_10.txt specimen a.karkala@nki.nl brush polygon`

1. This command writes the annotations of class "specimen" as written by user "a.karkala@nki.nl" for study 638 in .json format. 2. Type of annotations in this study are of two types: `brush `and `polygon`.  
3. Currently, we need to supply a text file of annotations that is manually downloaded from the slidescore study (`TISSUE_COMPARTMENTS_21_10_19_10.txt`).
4. While writing files to disc, the CLI adopts the following folder structure: `study_id/annotations/<slide_id>/<classname>.json`

### CODE REVIEW

**Comment by jonasteuwen** [comment]
@AjeyPaiK There are many failures in the CI/CD, can you address these first?

**Comment by jonasteuwen** [comment]
Ideally, you would want to have a command `convert-to-shapely` that does what you have. Also, why is the study required here?

I would have the `download-labels` support a `--raw` flag, that keeps the current behavior, and when you do not add this flag it exports it to shapely json's.

**Comment by jonasteuwen** [comment]
@AjeyPaiK Starts to look good. Can you describe the changes completely?

**Comment by AjeyPaiK** [comment]
@jonasteuwen , I'll edit the description in the PR.

**Review by EricMarcus-ai — COMMENTED** [review]
Overall, very nice code Ajey! It has been properly unspaghettified and is very readable. The only thing (except for some nitpick comments) I was thinking about is the possibility of asking for more than one segmentation class or question. I don't think it is possible right now; `question` is expected as a string everywhere, the quickstart only shows the option of writing one possibility, and I couldn't find anything that checks for multiple entries (I could be wrong of course). But you could imagine the case where you want to get annotations for e.g. 'foreground', 'tumor', 'stroma', etc.. Currently, you can just call it multiple times, but maybe it's nicer if you can just do it all at once?

File: docs/quickstart.rst [hunk 1]
```diff
@@ -7,23 +7,86 @@ If the :doc:`/installation` went smoothly, you should be able to run :bash:`slid
...
+                              Path to file with API token. Required if SLIDESCORE_API_KEY environment variable is not set. Will overwrite the environment variable if set.                                     (default: None)
+        -s STUDY_ID, --study STUDY_ID
+                              SlideScore Study ID (default: None)
+        --disable-certificate-check
+                              Disable the certificate check. (default: False)
+        --no-log              Disable logging. (default: False)
+        -v, --verbose         Verbosity level, e.g. -v, -vv, -vvv (default: 0)
+  
+  
+First things first
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply docs/quickstart.rst:33]: Wow @EricMarcus-ai! This is really good advice. I didn't realize that it's a bit of a hassle to look for proper commands until you pointed it out. I'll clean it up. 

I'm also looking into the multiple questions thing. I suppose we can change the input type to a list just like we do for the type of annotations. But I'll look at it in detail and rectify the issue.

File: slidescore_api/api.py [hunk 2]
```diff
@@ -250,15 +259,15 @@ def download_slide(
...
         shutil.move(str(temp_write_to), str(write_to))
 
         self._write_to_history(save_dir, write_to.name)
         return write_to
 
-    def get_results(self, study_id: int, **kwargs) -> List[SlideScoreResult]:
+    def get_results(self, study_id: int, **kwargs) -> Iterable[SlideScoreResult]:
         """
         Basic functionality to download all annotations made for a particular study.
         Returns a SlideScoreResult class wrapper containing the information.
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/api.py:273]: Cool, thanks for catching that!

File: slidescore_api/utils/annotations.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
+#!/home/ajey/miniconda3/bin/python3
+# coding=utf-8
+"""Utilility file containing parsing modules and functions to save slidescore annotations."""
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/utils/annotations.py:3]: Ohh, also, I should perhaps remove the shebangs!

File: slidescore_api/utils/annotations.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        `row_idx` need not be sequential and is mapped to the original idx in the text file.
+        Data is stored based on the type, accessed using `data["type"]`:
+            brush: type (str), positive_polygons (ndarray) , negative_polygons (ndarray)
+            polygon: type (str), points (ndarray)
+            ellipse: type (str), center (ndarray) , size (ndarray)
+        Empty rows have empty dict as data.
+
+        Parameters
+        ----------
+        row_iterator
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/utils/annotations.py]: This is done.

File: slidescore_api/utils/annotations.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/utils/annotations.py]: I've corrected the typo but not removed the shebangs.

File: slidescore_api/cli.py [hunk 5]
```diff
@@ -100,9 +114,11 @@ def _upload_labels(args: argparse.Namespace) -> None:
     client.upload_results(study_id, wsi_results)
 
 
-# TODO: This is how to actually retrieve the questions. Think about a proper way to do this.
 def retrieve_questions(
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/cli.py:117]: Fixed this.

File: slidescore_api/api.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/api.py:273]: I think someone already fixed it before me?

File: slidescore_api/cli.py [hunk 6]
```diff
@@ -131,16 +147,42 @@ def retrieve_questions(
...
+        yield annotation.to_row()
+
+
+def download_labels(  # pylint: disable=too-many-arguments,too-many-locals,too-many-branches
     slidescore_url: str,
     api_token: str,
     study_id: int,
     save_dir: Path,
+    output_type: str,
+    ann_type: list,
```
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply slidescore_api/cli.py:183]: Done. Thanks Eric. I'd never catch this. Quite scrupulous. :)

File: docs/quickstart.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (AjeyPaiK) [reply docs/quickstart.rst:33]: As it turns out, it's out of the scope of this repo to address the multiple-question problem. I looked up the slidescore API documentation and it turns out that it is only possible to consume only one string at a time. Click [here](https://www.slidescore.com/docs/api.html#:~:text=download%20scores%20from-,question,-string) for the documentation.

Perhaps, we should speak with Jan about this. @jonasteuwen

[the remark this answers is not in the document]
Comment (EricMarcus-ai) [reply docs/quickstart.rst]: Yes it seems that the multiple inputs is bit harder than I expected, we will get back to this later.

**Pull request merged**